### DISCUSSION TITLE
How to display the span id in a log message?

### DISCUSSION THREAD

**Discussion by zeodtr** [tokio-rs/tracing#2483] [Q&A] [answered]
It would be very helpful if the span id can be displayed on a log message that's generated by an event in a span that's created by an instrument macro. But I cannot find how to do it.

Without it, I must add an artificial id for the instrument macro instance, for example,

```rust
    #[instrument(skip_all, fields(id = %Uuid::new_v4()), err)]
    async fn a_function() {}
```

It would be nice if I can do like this:

```rust
    #[instrument(skip_all, span_id, err)]
    async fn a_function() {}
```

Or, it would be nice if `tracing_subscriber::fmt::SubscriberBuilder` has `with_span_id()` method.

Thanks in advance.

**Comment by davidbarsky** [comment, marked as answer]
> Without it, I must add an artificial id for the instrument macro instance.

What you're writing isn't really an artificial ID—it's an ID that makes sense for your application or library. `tracing`'s span IDs are generally meant to be opaque identifiers that are unique within a process for currently open spans. They are _not_ meant to be globally unique identifiers like UUIDs. If you'd like to read more, take a look at [this documentation](https://docs.rs/tracing/latest/tracing/trait.Subscriber.html#tymethod.new_span).

**Reply by zeodtr** [reply]
Thanks for your clear explanation.

**Comment by ahmedmedo256** [comment]
2023-02-23 19:41 غرينتش+02:00, David Barsky ***@***.***>:
